test/e2e/workflow/workflow.test.ts

```diff
@@ -0,0 +1,380 @@
+import WorkflowClient from "../../../src/workflow/client/WorkflowClient";
+import WorkflowContext from "../../../src/workflow/runtime/WorkflowContext";
+import WorkflowRuntime from "../../../src/workflow/runtime/WorkflowRuntime"
+import { TWorkflow } from "../../../src/workflow/types/Workflow.type";
+import { getFunctionName } from "../../../src/workflow/internal";
+import { WorkflowRuntimeStatus } from "../../../src/workflow/runtime/WorkflowRuntimeStatus";
+import WorkflowActivityContext from "../../../src/workflow/runtime/WorkflowActivityContext";
+import { Task } from "kaibocai-durabletask-js/task/task";
+
+
+describe("Workflow", () => {
+    const grpcEndpoint = "localhost:4001";
+    let workflowClient: WorkflowClient;
+    let workflowRuntime: WorkflowRuntime;
+
+    beforeAll(async () => { });
+
+    beforeEach(async () => {
+        // Start a worker, which will connect to the sidecar in a background thread
+        workflowClient = new WorkflowClient(grpcEndpoint);
```

Comment:
We are creating a new client and worker for each e2e test in all durabletask sdks (js, java, .net, etc) and the reason for it is to avoid interference between tests since many tests can use the same workflow name or ID.